lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1392,6 +1392,11 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
         return transValue(II->getOperand(0), BB);
     }
 
+    // Skip debug info for this call because this call is dropped in SPIR-V.
+    // TODO: remove skipping of debug info when *.annotation call will be
+    //       handled in a better way during IR <=> SPIR-V translation.
+    II->setDebugLoc(DebugLoc());
```

Comment:
I'm not sure this is the right place to insert this code. What if we return on line 1389 or 1392, the debug info will not be removed, right ?

Reply:
Debug info for annotation intrinsic line is dropped only in case if the whole call is dropped from SPIR-V.
On lines this call is translated into call of intel intrinsic fpga_reg or replaced by assignment of first argument of annotation method. So, in these cases debug info can be linked to a particular line in SPIR-V representation.

Reply:
Ok

Reply:
This fix is changed on more appropriate solution.